extensions/amp-auto-ads/0.1/ad-tracker.js

```diff
@@ -170,3 +177,71 @@ export function getExistingAds(ampdoc) {
         return true;
       });
 }
+
+/**
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {!Object<string, *>} configObj
+ * @return {?AdConstraints}
+ */
+export function getAdConstraintsFromConfigObj(ampdoc, configObj) {
+  if (!configObj['adConstraints']) {
+    return null;
+  }
+  const obj = configObj['adConstraints'];
+
+  const viewportHeight = Services.viewportForDoc(ampdoc).getSize().height;
+
+  const initialMinSpacing =
+      getValueFromString(obj['initialMinSpacing'], viewportHeight);
+  if (initialMinSpacing == null) {
+    user().warn(TAG, 'Invalid initial min spacing');
+    return null;
+  }
+
+  const subsequentMinSpacing = (obj['subsequentMinSpacing'] || []).map(item => {
+    const adCount = item['adCount'];
+    const spacing = getValueFromString(item['spacing'], viewportHeight);
+    if (spacing == null) {
+      user().warn(TAG, 'Invalid subsequent min spacing');
+      return null;
+    }
+    return {
+      adCount,
+      spacing,
+    };
+  });
+
+  if (subsequentMinSpacing.indexOf(null) != -1) {
+    return null;
+  }
+
+  return {
+    initialMinSpacing,
+    subsequentMinSpacing,
+    maxAdCount: obj['maxAdCount'],
```

Comment:
same here, do we need to check for `maxAdCount` is defined since it is required.